sigopt/objects.py

```diff
@@ -246,16 +246,27 @@ def iterate_pages(self):
     # pylint: disable=no-member
     data = self.data
     paging = self.paging or Paging({})
-    use_before = bool(paging.before)
+
+    # TODO(patrick): Probably should always rely on _retrieve_params vs. using paging.before,
+    # but that is divergent behaviour. Consider for next major version bump
+    specified_direction = ('before' in self._retrieve_params) ^ ('after' in self._retrieve_params)
+    if specified_direction:
+      use_before = 'before' in self._retrieve_params or 'after' not in self._retrieve_params
```

Comment:
Since `specified_direction` is set with `XOR`, probably don't need the `or after not in...` (but fine to leave for clarity)